src/pkg/label/manager.go

```diff
@@ -0,0 +1,94 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package label
+
+import (
+	"context"
+	"github.com/goharbor/harbor/src/common/models"
+	ierror "github.com/goharbor/harbor/src/internal/error"
+	"github.com/goharbor/harbor/src/pkg/q"
+	"time"
+)
+
+// Mgr is a global instance of label manager
+var Mgr = New()
+
+// Manager manages the labels and references between label and resource
+type Manager interface {
```

Comment:
label's CRUD and reference management should be two parts, right? 
label mgr for label CURD
labelMgr.ReferenceMgr() returns the ref mgr that is responsible for reference management?

Reply:
I'll submit another PR to cover this part for both label manager and artifact manager